projects/driver_controls/inc/stm32f0xx/ads1015.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+// Module for using ads1015
+// GPIO and Interrupt, I2C should be initiated
+// The user also needs to create a struct ADS1015Data which would persist accross functions
+// start the ads1015 using ads1015_init and then configure channels using ads1015_config_channel
+// read raw and converted values using ads1015_read_raw and ads1015_read_converted.
+#include <stdbool.h>
+#include "gpio.h"
+#include "i2c.h"
+#include "status.h"
+
+typedef enum {
+  ADS1015_ADDRESS_GND = 0,
+  ADS1015_ADDRESS_VDD,
+  ADS1015_ADDRESS_SDA,
+  ADS1015_ADDRESS_SCL,
+  NUM_ADS1015_ADDRESSES,
+} ADS1015Address;
+
+typedef enum {
+  ADS1015_CHANNEL_0 = 0,
+  ADS1015_CHANNEL_1,
+  ADS1015_CHANNEL_2,
+  ADS1015_CHANNEL_3,
+  NUM_ADS1015_CHANNELS,
+} ADS1015Channel;
+
+typedef void (*ADS1015Callback)(const GPIOAddress *address, void *context);
+
+typedef struct ADS1015Data {
+  I2CPort i2c_port;
+  ADS1015Address i2c_addr;
+  GPIOAddress *ready_pin;
```

Comment:
You might want to store a copy of this instead of a pointer if you need it again. (Of course, if you don't need it again, there's no point in keeping it)

Otherwise, imagine the user calls `ads1015_init()` with a stack-allocated ready pin.